### PR TITLE
Add text about TURN servers

### LINKED ISSUE #29: Don't pair mDNS candidates with relay candidates
If we resolve a name N1 to 1.1.1.1, should we pair it with relay candidates? If we do so, 1.1.1.1 will be sent to the TURN server, which seems like a leak (could be under the control of the web app).

Probably need to add a section in the privacy considerations saying any remote mdns candidates MUST NOT be paired with local relay candidates (although the reverse should be OK).

### PR DESCRIPTION
**Pull request by juberti**
Fixes #29

### CODE REVIEW

File: draft-ietf-rtcweb-mdns-ice-candidates.md [hunk 1]
```diff
@@ -251,6 +251,23 @@ same address and either the same or a different port; this includes cases
 where the signaled candidate is discarded as redundant according to 
 Section 5.1.3 of {{RFC8445}}.
 
+Interactions with TURN servers 
+------------------------------
+
+When sending data to a TURN {{RFC5766}} server, the sending client tells
+the server the destination IP and port for the data. This means that
+if the the client uses TURN to send to an IP that was obtained by mDNS
```
Comment (youennf) [inline draft-ietf-rtcweb-mdns-ice-candidates.md]: s/the the/the

File: draft-ietf-rtcweb-mdns-ice-candidates.md [hunk 2]
```diff
@@ -251,6 +251,23 @@ same address and either the same or a different port; this includes cases
...
+resolution, the TURN server will learn the underlying host IP and port,
+and this information can then be relayed to the web application,
+defeating the value of the mDNS wrapping. 
+
+To prevent disclosure of the host IP address to a TURN server, the ICE
+agent MUST NOT form candidate pairs between its own relay candidates
+and remote mDNS candidates. Note that the converse is not an issue; the
+ICE agent MAY form a candidate pairs between its own mDNS candidates and
```
Comment (youennf) [inline draft-ietf-rtcweb-mdns-ice-candidates.md]: s/a candidate pairs/candidate pairs/

File: draft-ietf-rtcweb-mdns-ice-candidates.md [hunk 3]
```diff
@@ -251,6 +251,23 @@ same address and either the same or a different port; this includes cases
...
+remote relay candidates, as in this situation host IPs will not be sent
+directly to the TURN server.
```
Comment (youennf) [inline draft-ietf-rtcweb-mdns-ice-candidates.md]: Interesting!
I am fuzzy of the practical impact of such a restriction.
It seems like it should not be limiting connectivity as the ICE agent can form candidate pairs between its relay candidates and remote self reflexive candidates.

File: draft-ietf-rtcweb-mdns-ice-candidates.md [hunk 4]
```diff
@@ -251,6 +251,23 @@ same address and either the same or a different port; this includes cases
 where the signaled candidate is discarded as redundant according to 
 Section 5.1.3 of {{RFC8445}}.
 
+Interactions with TURN servers 
+------------------------------
+
+When sending data to a TURN {{RFC5766}} server, the sending client tells
```
Comment (youennf) [inline draft-ietf-rtcweb-mdns-ice-candidates.md]: Might need to add a reference to RC5766 around line 40 of the document.

File: draft-ietf-rtcweb-mdns-ice-candidates.md (hunk 3 above)
Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: yeah, no practical impact.
Normally, you have:
- host->host, host->srflx, host->relay
- relay->host, relay->srflx, relay->relay

relay->host is what we're dropping, but I can't think of any scenario where that would work when the host candidate is private.

File: draft-ietf-rtcweb-mdns-ice-candidates.md (hunk 4 above)
Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: I think I did this...

File: draft-ietf-rtcweb-mdns-ice-candidates.md (hunk 3 above)
Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: I could mention that this has no practical impact.

Comment (juberti) [reply draft-ietf-rtcweb-mdns-ice-candidates.md]: Paragraph added.

**Comment by youennf** [comment]
LGTM too, will resolve conflicts

**Pull request merged**